### PR TITLE
23832876 SearchBooksByTitle

### PR DESCRIPTION
**Pull request by tinkabel85**

### CODE REVIEW

File: src/Controller/BookController.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  public function searchBooksByTitle(Request $request): Response
+  {
+    $query = $request->query->get('query');
+    $apiKey = getenv('BOOKS_APP_API_KEY');
+    $url = "https://www.googleapis.com/books/v1/volumes?q=?+intitle:$query&key=$apiKey";
+    $result = $this->curlService->setStrategy(new CurlGetStrategy())->setUrl($url)->doRequest();
+
+    return $this->json(['result' => json_decode($result)]);
+  }
```
Comment (anna-bc) [inline src/Controller/BookController.php:38]: I would get the title from the url. So if we use /books/{title} the code would look like this:

```
public function searchBooksByTitle(string $title): Response
  {
    $apiKey = getenv('BOOKS_APP_API_KEY');
    $url = "https://www.googleapis.com/books/v1/volumes?q=?+intitle:$title&key=$apiKey";
    $result = $this->curlService->setStrategy(new CurlGetStrategy())->setUrl($url)->doRequest();

    return $this->json(['result' => json_decode($result)]);
  }
```

File: templates/book/index.html.twig [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block title %}Books{% endblock %}
+
+{% block body %}
+<style>
+</style>
+
+<div>
+    <h1>The list of the books</h1>
+</div>
```
Comment (anna-bc) [inline templates/book/index.html.twig:11]: we need this, as we will use React

File: config/routes.yaml [hunk 3]
```diff
@@ -3,3 +3,7 @@ controllers:
         path: ../src/Controller/
         namespace: App\Controller
     type: attribute
+
+curl_search:
+    path: /books/searchByTitle
+    controller: App\Controller\BookController::searchBooksByTitle
```
Comment (anna-bc) [inline config/routes.yaml:9]: change path to /books/{title}

**Review by anna-bc — CHANGES_REQUESTED** [review]

File: config/routes.yaml (hunk 3 above)
Comment (tinkabel85) [reply config/routes.yaml:9]: done!

**Review by anna-bc — APPROVED** [review]

**Review by naomiinanna — APPROVED** [review]

**Pull request merged**